File: desh/ege2026kp/24solve/24-160.pas
// Автор В.Н. Шубинкин
##
var s := ReadLines('24-s1.txt').minBy(line -> line.count(c -> c = 'A'));
var max_letter_count := 0;
var letter := '';
for var let := 'A' to 'Z' do
begin
  var let_count := s.Count(c -> c = let);
  Println(let, let_count);
  if let_count >= max_letter_count then
  begin
    max_letter_count := let_count;
    letter := let
  end;
end;
write(letter, ReadAllText('24-s1.txt').Count(t -> t = letter))

// Вместо Count(c -> c = let) можно было бы воспользоваться CountOf(let),
// но на экзамене может быть установлена недостаточно свежая версия PascalABC.NET

File: desh/ege2026kp/24solve/24-160.py
sMinA, fullText = "", ""
for s in open("24-s1.txt"):
  if not sMinA or s.count('A') < sMinA.count('A'):
     sMinA = s
  fullText += s

from string import ascii_uppercase

mostFreq = None
for c in ascii_uppercase:
  freq = sMinA.count(c)
  if not mostFreq or freq >= mostFreq[1]:
    mostFreq = (c, freq)

print( mostFreq[0], fullText.count(mostFreq[0]) )

"""
ABC = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
with open("24-s1.txt") as F:
  data = F.readlines()
  minK = 10**10
  for s in data:
    k = s.count('A')
    if k < minK:
      minK = k
      ma, letterMax = 0, ''
      for letter in ABC:
        cnt = s.count(letter)
        if cnt >= ma:
          ma, letterMax = cnt, letter

print( letterMax,
       sum(s.count(letterMax) for s in data) )
"""